src/ui/components/SearchCoordinates/component.ts

```diff
@@ -0,0 +1,32 @@
+import Component, { tracked } from '@glimmer/component';
+
+export default class SearchForm extends Component {
+  args: {
+    onSearch: (coordinates: Promise<number[]>) => void;
+  };
+
+  searchByLocation(event) {
+    if (event) {
+      event.preventDefault();
+    }
+
+    let { onSearch } = this.args;
+
+    let coordinatesPromise: Promise<number[]> = new Promise((resolve, reject) => {
```

Comment:
Can we use `async`/`await` here instead of a promise?

Reply:
We use `async/await` to consume the Promise, but here we're creating it.

From what I understand, `async/await` is a structure to handle promises. Though, I'm not aware of how `async/await` can be used to replace promises. Couldn't find any reference online on this direction. Maybe I'm missing something?

Reply:
Yeah, nevermind - I'm not sure what I was thinking here 😞